frontend/main.js

```diff
@@ -1,13 +1,74 @@
 const { app, BrowserWindow, dialog, ipcMain } = require("electron/main");
 const path = require("node:path");
-
+const fs = require("fs-extra");
+const { PythonShell } = require("python-shell");
+const constants = require("./constants/constants.js");
 let mainWindow;
 
+function runProgram(filePath) {
+  mainWindow.loadFile("loading.html");
+
+  const script = constants.SCRIPT;
+
+  const poetryArgs = [
+    constants.COMMANDS.DEDUPE_DATA,
+    constants.OPTIONS.FORMAT,
+    constants.FORMAT.TEST,
+    filePath,
+    "./",
+  ];
+
+  let options = {
+    mode: "text",
+    scriptPath: "../cli",
+    pythonPath: "../.venv/bin/python",
```

Comment:
Similar to the comment about using `__file__` for Python, there's `__filename` for Node that we might want to use here

Reply:
might also help prevent path traversal attacks?